src/Fixer/ClassNotation/StaticPrivateMethodFixer.php

```diff
@@ -0,0 +1,308 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\ClassNotation;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+use PhpCsFixer\Tokenizer\TokensAnalyzer;
+
+/**
+ * @author Filippo Tessarotto <zoeslam@gmail.com>
+ */
+final class StaticPrivateMethodFixer extends AbstractFixer
+{
+    /**
+     * @var array
+     */
+    private $magicNames = [
+        '__call' => true,
+        '__callstatic' => true,
+        '__clone' => true,
+        '__construct' => true,
+        '__debuginfo' => true,
+        '__destruct' => true,
+        '__get' => true,
+        '__invoke' => true,
+        '__isset' => true,
+        '__serialize' => true,
+        '__set' => true,
+        '__set_state' => true,
+        '__sleep' => true,
+        '__tostring' => true,
+        '__unserialize' => true,
+        '__unset' => true,
+        '__wakeup' => true,
+    ];
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Converts private methods to `static` where possible.',
+            [
+                new CodeSample(
+                    '<?php
+class Foo
+{
+    public function bar()
+    {
+        return $this->baz();
+    }
+
+    private function baz()
+    {
+        return 1;
+    }
+}
+'
+                ),
+            ],
+            null,
+            'Risky when method contains dynamic generated calls to the instance, or the method is dynamically referenced.'
```

Comment:
This is also risky if the private method is referenced from a trait, or a private method from a trait is referenced, as i don't think the fixer can pick up on that.

Reply:
Thank you, added